auto_cpufreq/core.py

```diff
@@ -1119,16 +1070,9 @@ def sysinfo():
                 core = cpu_core[cpu]
                 cpu_temp_index = core_temp_labels.index(f"Core {core}")
                 temp_per_cpu[i] = core_temp["coretemp"][cpu_temp_index].current
-        elif "k10temp" in core_temp:
-            # https://www.kernel.org/doc/Documentation/hwmon/k10temp
-            temp_per_cpu = [core_temp["k10temp"][0].current] * online_cpu_count
-        elif "zenpower" in core_temp:
-            # https://github.com/AdnanHodzic/auto-cpufreq/issues/145#issuecomment-763294009
-            temp_per_cpu = [core_temp["zenpower"][0].current] * online_cpu_count
-        elif "acpitz" in core_temp:
-            temp_per_cpu = [core_temp["acpitz"][0].current] * online_cpu_count
-        elif "thinkpad" in core_temp:
-            temp_per_cpu = [core_temp["thinkpad"][0].current] * online_cpu_count
+        else:
+            temp = list(psutil.sensors_temperatures())
+            temp_per_cpu = [core_temp[temp[0]][0].current] * online_cpu_count
     except Exception as e:
```

Comment:
I made it more generic using the same princip like at the fan speed display

Reply:
Good idea, I just hope it remains working on all of that different hardware, let's keep an eye on it based on new issues that come in.

Reply:
I am pretty sure it should work.
I just changed it so it takes the name to get the tame instead of checking for the name to get the temp

Reply:
Let's go for it, but let's also just keep an eye on new issues and if they could be related to changes here.